### DISCUSSION TITLE
x-model on custom element (web component) made using Vue

### DISCUSSION THREAD

**Discussion by MichelJonkman** [alpinejs/alpine#4256] [1. Help]
I've made a custom form element using Vue 3 web components and I'm trying to use the x-model directive on it but I can't figure out how to tell Alpine what the value should be! 

My idea was to use events for this but I can't figure out what event I could use that lets me pass a value easily.

Could anyone help me out with this?

**Comment by ekwoka** [comment]
the change event, just like native form inputs.

**Reply by MichelJonkman** [reply]
I noticed the change event but couldn't figure out where Alpine was pulling the value from

**Reply by ekwoka** [reply]
from the `value` property for nonstandard inputs.

So you emit a change event, and expose a value property.

**Reply by MichelJonkman** [reply]
Could you send me an example? I'm not sure what you mean by "nonstandard inputs"

**Reply by ekwoka** [reply]
non-native I guess?

like your custom element.
